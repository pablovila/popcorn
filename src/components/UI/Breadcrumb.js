import React from "react";
import PropTypes from "prop-types";
import { Link } from "server/routes";

const breadcrumb = ({ levels }) => {
  const numLevels = levels.length;
  return (
    <>
      <nav className="breadcrumb" aria-label="breadcrumbs">
        <ul>
          {levels.map((level, i) => (
            <li key={i} className={i + 1 === numLevels ? "is-active" : ""}>
              <Link route={level.route} params={level.params}>
                <a href={level.href}>{level.name}</a>
              </Link>
            </li>
          ))}
        </ul>
      </nav>
    </>
  );
};

breadcrumb.propTypes = {
  levels: PropTypes.array.isRequired
};

export default breadcrumb;
