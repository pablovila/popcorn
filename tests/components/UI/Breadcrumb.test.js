import React from "react";
import { shallow } from "enzyme";

import Breadcrumb from "components/UI/Breadcrumb";

describe("test Breadcrumb component", () => {
  it("renders Breadcrumb component", () => {
    const breadcrumb = shallow(<Breadcrumb levels={[]} />);

    expect(breadcrumb.find(".breadcrumb")).toHaveLength(1);
    expect(breadcrumb).toMatchSnapshot();
  });

  it("renders breadcrumb children and UI subcomponents", () => {
    const levels = [
      {
        name: "Genres",
        route: "/"
      },
      {
        name: "Action",
        route: "movies",
        href: "/movies/1",
        params: { slug: 1 }
      },
      {
        name: "Captain Marvel",
        route: "movie",
        href: `/movies/1/1`,
        params: { slug: 1, genreId: 1 }
      }
    ];

    const breadcrumb = shallow(<Breadcrumb levels={levels} />);
    expect(breadcrumb.find(".breadcrumb")).toHaveLength(1);
    expect(breadcrumb.find("ul")).toHaveLength(1);
    expect(breadcrumb.find("li")).toHaveLength(3);
    expect(breadcrumb.find("li.is-active")).toHaveLength(1);
    expect(breadcrumb.find("a")).toHaveLength(3);
  });
});
