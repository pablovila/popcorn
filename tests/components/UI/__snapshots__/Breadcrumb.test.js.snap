// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Breadcrumb component renders Breadcrumb component 1`] = `ShallowWrapper {}`;
